# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a=101
b=210
c=111
d=402
e=(a+b)*c/d
print('Результат виконання -', e)

Результат виконання - 85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [5]:
f=11283
j=89
k=a%b
print('Остача від ділення числа', f,'на', j,':', k)

Остача від ділення числа 11283 на 89 : 69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [6]:
g=f//j
print('Цілочисельне ділення',f,'на',j,'дає в результаті',g)

Цілочисельне ділення 11283 на 89 дає в результаті 126


4. Запишіть у змінну `name` ваше імʼя.

In [7]:
name='Oleksandra'

5. Запишіть у змінну `surname` ваше прізвище.

In [8]:
surname='Pidhorna'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [9]:
name_surname=name+' '+surname
print(name_surname)

Oleksandra Pidhorna


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [10]:
upper_name=name_surname.upper()

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [11]:
print(upper_name)

OLEKSANDRA PIDHORNA


8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [12]:
print('Третя літера імені -', name_surname[2])

Третя літера імені - e


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [13]:
print('Перша літера прізвища -', surname[0],', а остання -', surname[-1])

Перша літера прізвища - P , а остання - a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [14]:
print(len(surname))

8


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [15]:
cap='Antananarivo'
cap_low=cap.lower()
count_a=cap_low.count("a")
print('Кількість літер "a" у назві столиці Мадагаскару -', count_a)

Кількість літер "a" у назві столиці Мадагаскару - 4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [17]:
my_string="The quick brown fox jumps over the lazy dog!!!"
print('Чи рядок починається з "The":', my_string.startswith("The"))
print('Чи рядок закінчується на  "dog!":', my_string.endswith("dog!"))

Чи рядок починається з "The": True
Чи рядок закінчується на  "dog!": False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [19]:
my_string_list=my_string.split()
my_new_string='_'.join(my_string_list)
print(my_new_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`